beacon-chain/sync/database.go

```diff
@@ -0,0 +1,16 @@
+package sync
+
+import eth "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+
+type BlobSidecarGetter interface {
+	BlobSidecar(blockRoot [32]byte, index uint64) (*eth.BlobSidecar, error)
+}
+
+type BlobSidecarWriter interface {
+	WriteBlobSidecar(blockRoot [32]byte, index uint64, sidecar *eth.BlobSidecar) error
+}
+
+type BlobDB interface {
+	BlobSidecarGetter
+	BlobSidecarWriter
+}
```

Comment:
there is no concrete implementation of this interface, so while beacon-chain currently builds, nothing satisfying this interface is given to the sync service at init time, so methods that use this (like blobs by root) will throw a nil pointer exception.